libs/MVS/SceneTexture.cpp

```diff
@@ -535,6 +542,9 @@ bool MeshTexture::ListCameraFaces(FaceDataViewArr& facesDatas, float fOutlierThr
 				ASSERT((idxFace == NO_ID && depthMap(j,i) == 0) || (idxFace != NO_ID && depthMap(j,i) > 0));
 				if (idxFace == NO_ID)
 					continue;
+				if (!mask.empty() && !mask.isSet(j, i)) {
```

Comment:
this is not ok, there will be cases where the face partially overlaps the invalid region of the mask, so the few pixels that remain valid will assign the view to the mask even though the face contains invalid pixels; pls see PR #968 for a correct implementation; pls build on top of it; if so, pls move the mask estimation in that PR ourside under a command line paramter, to be estimated if the scene does not contain mask files.

Reply:
Ok, I understand the right way to implement it, I will finish it.